src/app.js

```diff
@@ -1,21 +1,20 @@
 'use strict';
 
-/**
- * Implement sum function:
- *
- * Function takes 2 numbers and returns their sum
- *
- * sum(1, 2) === 3
- * sum(1, 11) === 12
- *
- * @param {number} a
- * @param {number} b
- *
- * @return {number}
- */
-function sum(a, b) {
-  // write code here
-  return a + b;
-}
+const http = require('http');
 
-module.exports = sum;
+const server = http.createServer((req, res) => {
+  res.setHeader('Content-Type', 'application/json');
+
+  const normalizedURL = new URL(req.url, `http//${req.headers.host}`);
+  const parts = normalizedURL.pathname.slice(1).split('/');
+  const query = Object.fromEntries(normalizedURL.searchParams.entries());
```

Comment:
Unfortunately, this method doesn't work with duplicate search params ex: `/goods/phone?name=iphone&name=samsung`
As a suggestion we can use `searchParams` method `forEach()` and just group into one array the values that under one param.
```suggestion
  const query = {};
  
  normalizedURL.searchParams.forEach((value, key) => {
    if (!query[key]) {
      query[key] = value;
    } else {
      if (!Array.isArray(query[key])) {
        query[key] = [query[key]];
      }

      query[key].push(value);
    }
  });
```

Reply:
agree